docs/Additional_Build_Documentation.md

```diff
@@ -0,0 +1,296 @@
+# Building Benchmark
+
+To build Benchmark we need to follow a series of simple steps:
+*   Obtain Benchmark
+*   Configure Benchmark
+*   Build Benchmark
+
+To obtain Benchmark we can simply clone the Github repository using
+```bash
+$ git clone https://github.com/google/benchmark.git
+```
+Now, Benchmark uses Google Test to run it's tests. So, if we wish to build and run
+the tests we need to include googletest, which can be done either by cloning the googletest
+repository or by asking benchmark in the configuration step to download this dependency.
+We can clone the googletest repository in benchmark using
+```bash
+$ git clone https://github.com/google/googletest.git benchmark/googletest
+```
+To configure Benchmark we use CMake as follows:
+```bash
+$ cmake -G <generator> [options] -S benchmark -B build
+```
+If we haven't cloned the googletest repository as described above we can ask benchmark to download this dependency in the above configuration step by adding `-DBENCHMARK_DOWNLOAD_DEPENDENCIES=ON`
+
+Then to build Benchmark we use CMake again as follows:
+```bash
+$ cmake -build build
```

Comment:
alternatively, users can use cmake to generate and then Visual Studio or whatever build system they want to build.

Reply:
yeah, will add that. I feared that might create a situation where we have to give instructions for each generator. but will add it.

Reply:
added example for VS 15 2017, the build documentation in `README.md` was good and so added the example there. and removed the build documentation form the file added by me.